api/py/ai/chronon/repo/validator.py

```diff
@@ -322,7 +384,10 @@ def _validate_group_by(self, group_by: GroupBy) -> List[str]:
 
         # validate the derivations are defined correctly
         if group_by.derivations:
-            columns = set(get_pre_derived_group_by_columns(group_by))
+            columns = get_pre_derived_group_by_columns(group_by)
```

Comment:
nit: put this in else condition to avoid repeated calculation, because most gb are online

Reply:
updated